src/components/lesson/data.ts

```diff
@@ -1,17 +1,50 @@
+
+export type FormItemData = {
+  itemId?: string;
+  displayName?: string | React.ReactNode;
+  description?: string;
+  format?: "question" | "image" | "embed" | "text";
+  questionType?: "radio" | "checkbox" | "drop down" | "file upload" | "text" | "date" | "time";
+  options?: string[];
+  paragraph?: boolean;
+  required?: boolean;
+  reflection?: string;
+  rating?: number;
+  disabled?: boolean;
+  url?: string;
+  answerId?: string;
+  journalId?: string;
+  questionId?: string;
+  responses?: string[];
+  minText?: number;
+  children?: React.ReactNode;
+
+};
+
+
 /**
  * LessonData
  */
 export type LessonData = {
-  title: string
-  href: string;
-  isLiked: boolean;
-  isSaved: boolean;
-  numberOfLikes: number;
-  description: string
-  estimate: string
-  tags: string[]
-  rating: number
-  preview: {title: string}[]
-  numberOfQuestions: number
-  uploadedOn: string
-}
\ No newline at end of file
+  formId?:string;
+  title?: string
+  href?: string;
+  imageURL?:string;
+  isLiked?: boolean;
+  isSaved?: boolean;
+  numberOfLikes?: number;
+  description?: string
+  estimate?: string
+  tags?: string[]
+  rating?: number
+  preview?: {title: string}[]
+  numberOfQuestions?: number
+  uploadedOn?: string
+  reflection?: string;
+  items?: FormItemData[];
+  timeSpent?: number;
+  elapsedTime?: number;
+  stopWatchStarted?: boolean;
+  stopWatch?: React.ReactNode;
```

Comment:
We should only use serializable types in our data models (if you need React.ReactNode you should define it in the component properties). Consider data as something (that can be persisted and retrieved). For example,

`
type ComponentProps = ComponentData & {
   children?: React.ReactNode
   myCallback?: () => void;
}
`

Reply:
have updated the data model